lib/exception_handling.rb

```diff
@@ -220,7 +220,13 @@ def log_error(exception_or_string, exception_context = '', controller = nil, tre
     #
     def write_exception_to_log(ex, exception_context, timestamp, log_context = {})
       ActiveSupport::Deprecation.silence do
-        ExceptionHandling.logger.fatal("\nExceptionHandlingError (Error:#{timestamp}) #{ex.class} #{exception_context} (#{encode_utf8(ex.message.to_s)}):\n  " + clean_backtrace(ex).join("\n  ") + "\n\n", log_context)
+        log_message = "\nExceptionHandlingError (Error:#{timestamp}) #{ex.class} #{exception_context} (#{encode_utf8(ex.message.to_s)}):\n  " + clean_backtrace(ex).join("\n  ") + "\n\n"
```

Comment:
are we okay with `ExceptionHandlingError` and `(Error:...` iin the log_message or should we update those as well?

Maybe this?
```
"\nExceptionHandlingWarning (Warning:#{timestamp}..."
```

Reply:
I'm a little torn on this. I think these are actually errors that we're logging, they just weren't fatal to the execution of the program? I could go either way.

If they aren't the same thing, maybe log_warning shouldn't be calling log_error? I wouldn't mind attempting a slight refactor to separate what log_error is responsible for.

Reply:
We've decided to internally use `log_error` when calling `log_warning`.. but that is just an implementation detail. From the caller's perspective, we are doing a log_warning, so I think it will be least surprising if the text that is put in the log (and kibana) shows up as:

```
"\nExceptionHandlingWarning (Warning:#{timestamp}..."
```

As you hinted at, it could mean a bit of a refactor is needed to pull this logic apart.. on the other hand you already have the `ex.is_a?(Warning)` check below.. so could definitely move some of those pieces down there too:

```
log_message = "#{exception_context}\n#{ex.class}: #{encode_utf8(ex.message.to_s)}\n  " + clean_backtrace(ex).join("\n  ") + "\n\n"

if ex.is_a?(Warning)
  ExceptionHandling.logger.warn("\nExceptionHandlingWarning (Warning:#{timestamp}) #{log_message}", log_context)
else
  ExceptionHandling.logger.fatal("\nExceptionHandlingError (Error:#{timestamp}) #{log_message}", log_context)
end
```

(BTW the above example incorporates Colin's [suggestion from Slack](https://invoca.slack.com/archives/CRGB17J59/p1597887172000700) too)